app_dart/lib/src/service/github_status_service.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../model/luci/buildbucket.dart' as bb;
+import '../model/luci/push_message.dart';
+import 'luci_build_service.dart';
+
+class GithubStatusService {
+  const GithubStatusService(this.config, this.luciBuildService);
+
+  /// The global configuration of this AppEngine server.
+  final Config config;
+  final LuciBuildService luciBuildService;
+
+  Future<void> setBuildsPendingStatus(
+      String repositoryName, int prNumber, String commitSha) async {
+    final RepositorySlug slug = RepositorySlug('flutter', repositoryName);
+    final GitHub gitHubClient = await config.createGitHubClient();
+    final Map<String, bb.Build> builds = await luciBuildService
+        .buildsForRepositoryAndPr(repositoryName, prNumber, commitSha);
+    for (bb.Build build in builds.values) {
+      final CreateStatus status = CreateStatus('pending')
+        ..context = build.builderId.builder
+        ..description = 'Flutter LUCI Build: ${build.builderId.builder}'
+        ..targetUrl = '';
+      await gitHubClient.repositories.createStatus(slug, commitSha, status);
+    }
+  }
+
+  Future<void> setPendingStatus({
+    @required String ref,
+    @required String builderName,
+    @required String buildUrl,
+  }) async {
+    final RepositorySlug slug = await config.repoNameForBuilder(builderName);
+    // No builderName configuration, nothing to do here.
+    if (slug == null) {
+      return;
+    }
```

Comment:
Without logging, it would be hard to debug when slug is null as this function just returns sliently.

Reply:
This class runs outside of AppEngine context and logging is delegated to the handler class calling the service.

Reply:
Fixed it.